src/compas_timber/ghpython/components/CT_Assembly/code.py

```diff
@@ -68,26 +67,27 @@ def process_joint_rules(self, beams, rules, max_distance=0):
                         continue
                     if rule.joint_type.SUPPORTED_TOPOLOGY != detected_topo:
                         msg = "Conflict detected! Beams: {}, {} meet with topology: {} but rule assigns: {}"
-                        Info.append(
-                            msg.format(beam_a, beam_b, JointTopology.get_name(detected_topo), rule.joint_type.__name__)
+                        self.AddRuntimeMessage(Warning,
```

Comment:
I cant figure out why the beam.key returns None

Reply:
beams only get a key once added to an assembly. it's a graph thing.

Reply:
yes but I pass the beams from `Assembly.beams` to this function. see lines 118 and 120

Reply:
Ok i resolved this